### PR TITLE
Switch select() to poll() in pty_read()

### PR DESCRIPTION
**Pull request by arjanvandervelde**
This fixes https://github.com/jupyter/terminado/issues/110

### CODE REVIEW

**Comment by blink1073** [comment]
Kicking CI

**Comment by blink1073** [comment]
Not sure why CI isn't running, but I'd like it to run because of the "Not supported by all operating systems" note in the [official docs](https://docs.python.org/3/library/select.html#select.poll).  I'll try kicking the build again tomorrow.

**Comment by blink1073** [comment]
Indeed, the Windows builds are failing with `AttributeError: module 'select' has no attribute 'poll'`

**Comment by arjanvandervelde** [comment]
Ok, I'll update the PR to account for that.

File: terminado/management.py [hunk 1]
```diff
@@ -206,8 +219,9 @@ def on_eof(self, ptywclients):
 
     def pty_read(self, fd, events=None):
         """Called by the event loop when there is pty data ready to read."""
-        r, _, _ = select.select([fd], [], [], .1)
-        if not r:
+        # prevent blocking on fd
+        if not _poll(fd, timeout=0.1):  # 100ms
+            self.log.info(f"Spurious pty_read() on fd {fd}")
```
Comment (afshin) [inline terminado/management.py:224]: Hello! Could you please log the output via `.debug(...)` here instead of `.info(...)`?

Comment (arjanvandervelde) [reply terminado/management.py]: done.

**Review by blink1073 — APPROVED** [review]
Nice work, thank you @arjanvandervelde!

**Pull request merged**